sentinel-extension/sentinel-annotation-aspectj/src/main/java/com/alibaba/csp/sentinel/annotation/aspectj/AbstractSentinelAspectSupport.java

```diff
@@ -149,21 +133,49 @@ protected Object handleBlockException(ProceedingJoinPoint pjp, SentinelResource
             // Construct args.
             Object[] args = Arrays.copyOf(originArgs, originArgs.length + 1);
             args[args.length - 1] = ex;
-            try {
-                if (isStatic(blockHandlerMethod)) {
-                    return blockHandlerMethod.invoke(null, args);
-                }
-                return blockHandlerMethod.invoke(pjp.getTarget(), args);
-            } catch (InvocationTargetException e) {
-                // throw the actual exception
-                throw e.getTargetException();
-            }
+            return invoke(pjp, blockHandlerMethod, args);
         }
 
         // If no block handler is present, then go to fallback.
         return handleFallback(pjp, annotation, ex);
     }
 
+    private Object invoke(ProceedingJoinPoint pjp, Method method, Object[] args) throws Throwable {
+        boolean accessible = method.isAccessible();
+        try {
+            if (!accessible) {
+                makeAccessible(method);
+            }
+            if (isStatic(method)) {
+                return method.invoke(null, args);
+            }
+            return method.invoke(pjp.getTarget(), args);
+        } catch (InvocationTargetException e) {
+            // throw the actual exception
+            throw e.getTargetException();
+        } finally {
+            if (!accessible) {
```

Comment:
@wuwen5 It maybe have a concurrent problem here. the method's accessiable flag maybe change to false before one thread execute its method.invoke. Could you test you codes in the multi threads environment?

Reply:
If there have no concurrent problem, I think the codes in finally is unnecessary. do you think so?